src/modules/ctc/inc/ctc.h

```diff
@@ -7,28 +7,162 @@
 #define TRAINS_SRC_MODULES_CTC_INC_CTC_H
 
 #include <unordered_map>
+#include <algorithm>
+#include <filesystem>
 
 #include "types.h"
+#include "csv_parser.h"
+#include "block_builder.h"
 #include "wayside_controller.h"
 #include "wayside_controller_gateway.h"
+#include "graph.h"
 
 namespace ctc
 {
 
-typedef struct
+typedef uint16_t                         column;
+typedef struct DestinationAndArrivalTime DestinationAndArrivalTime;
+typedef struct Train                     Train;
+typedef struct Station                   Station;
+
+typedef enum
+{
+    SectionDirection_Forward,
+    SectionDirection_Reverse,
+} SectionDirection;
+
+typedef enum
+{
+    AUTOMATIC_MODE,
+    MANUAL_MODE,
+    MAINTENANCE_MODE
+} CtcOperationMode;
+
+struct Station
```

Comment:
To have a vector of stations as data member of ctc class. This vector is used to display a combo box on the ui for user to select destination so that Riley doesn't have to remember all stations' name